torch/distributed/fsdp/flat_param.py

```diff
@@ -566,6 +571,12 @@ def _init_flat_param_and_metadata(
             _convert_to_params(shared_params) if use_orig_params else None,
             is_padding_mask,
         )
+        if self._use_orig_params:
+            assert self.flat_param._params is not None  # mypy
+            for param, requires_grad in zip(
+                self.flat_param._params, requires_grad_mask
+            ):
+                param.requires_grad = requires_grad
```

Comment:
why do we need to set this here? If these are the original params, shouldn't this value already be populated?

Reply:
Good point!